server/src/completion.ts

```diff
@@ -148,10 +148,15 @@ export function asCommitCharacters(kind: ScriptElementKind): string[] | undefine
 export function asResolvedCompletionItem(item: TSCompletionItem, details: tsp.CompletionEntryDetails): TSCompletionItem {
     item.detail = asDetail(details);
     item.documentation = asDocumentation(details);
+    item.deprecated = isDeprecated(details.tags);
     Object.assign(item, asCodeActions(details, item.data.file));
     return item;
 }
 
+function isDeprecated(tags?: tsp.JSDocTagInfo[]): boolean | undefined {
+    return tags?.some(tag => tag.name.toLowerCase() === 'deprecated');
```

Comment:
If I recall correctly, you can check if a TypeScript completion item is deprecated by checking the `TSCompletionItem.kindModifiers` (since version 4).

https://github.com/microsoft/TypeScript/pull/38523/files#diff-ac54f9219619989a94c776b937ce1ebcbcc7dc6d7c424effab113f2ed0eb016fR19

You can see this [PR](https://github.com/microsoft/TypeScript/pull/38523/files#diff-ac54f9219619989a94c776b937ce1ebcbcc7dc6d7c424effab113f2ed0eb016fR19) that added the deprecated support in TS for more info. 

-----
I think that it would be good to rely on `kindModifiers` for workspace that has TS version 4 and above.

For workspace(project) that use a version of typescript below version 4. 
I would not try to put additional effort to mark the completion item as deprecated (because TS didn't support it prior version 4).

But If you really want to mark completion items as deprecated, you can use this approach that you already implemented. (by checking the tags in the docs comment).

Reply:
I haven't tried this PR, 

but I think performance would be affected when trying to traverse the docs comment for each completion item(CI). So the completion response might need (a lot more, depending on the project) time to compute and return. (performance was also a [concern](https://github.com/microsoft/TypeScript/pull/38523#issuecomment-645528105) when they implemented it in TS)

That is the reason why I would rely only on `kindModifiers`.

Reply:
Changed the code to use `kindModifiers` and without having to resolve item.